modules/tasks/tasks/src/main/java/org/motechproject/tasks/json/TaskDeserializer.java

```diff
@@ -64,7 +72,13 @@ public Task deserialize(JsonParser jsonParser,
         setProperty("enabled", stringType);
         setProperty("hasRegisteredChannel", stringType);
         setProperty("taskConfig", typeFactory.constructType(TaskConfig.class));
-        setProperty("trigger", typeFactory.constructType(TaskTriggerInformation.class));
+
+        if (jsonNode.get("trigger").toString().contains(SCHEDULER_TASK)) {
```

Comment:
I'm pretty sure there is a better way to do this using type discriminators, but it can stay like this for now